analysis/export/CSVExporter/src/test/kotlin/de/maibornwolff/codecharta/exporter/csvexporter/CSVExporterTest.kt

```diff
@@ -3,37 +3,239 @@ package de.maibornwolff.codecharta.exporter.csvexporter
     }
 
     @Test
-    fun `should stop execution if input files are invalid`() {
+    fun `should create correct output for single valid file as input source`() {
+        val inputFilePath = "src/test/resources/input_valid_1.cc.json"
+        val outputFilePath = "src/test/resources/output.csv"
+
+        val outputFile = File(outputFilePath)
+        val referenceFile = File("src/test/resources/reference_valid_1.csv")
+        outputFile.deleteOnExit()
+
+        CommandLine(CSVExporter()).execute(inputFilePath, "-o", outputFilePath)
+
+        Assertions.assertThat(outputFile.exists()).isTrue()
+        Assertions.assertThat(outputFile.length()).isNotEqualTo(0)
+        Assertions.assertThat(outputFile).hasSameTextualContentAs(referenceFile)
+    }
+
+    @Test
+    fun `should create correct output for two valid files as input sources`() {
+        val inputFilePath1 = "src/test/resources/input_valid_1.cc.json"
+        val inputFilePath2 = "src/test/resources/input_valid_2.cc.json"
+        val outputFilePath = "src/test/resources/output.csv"
+
+        val outputFile = File(outputFilePath)
+        val referenceFile = File("src/test/resources/reference_valid_1_valid_2.csv")
+        outputFile.deleteOnExit()
+
+        CommandLine(CSVExporter()).execute(inputFilePath1, inputFilePath2, "-o", outputFilePath)
+
+        Assertions.assertThat(outputFile.exists()).isTrue()
+        Assertions.assertThat(outputFile.length()).isNotEqualTo(0)
+        Assertions.assertThat(outputFile).hasSameTextualContentAs(referenceFile)
+    }
+
+    @Test
+    fun `should fail to create output for invalid file as input source`() {
+        val invalidInputFilePath = "filePathDoesNotExist.cc.json"
+
         mockkObject(InputHelper)
         every {
             InputHelper.isInputValid(any(), any())
         } returns false
 
         System.setErr(PrintStream(errContent))
-        CommandLine(CSVExporter()).execute("thisDoesNotExist.cc.json").toString()
+        CommandLine(CSVExporter()).execute(invalidInputFilePath)
+        System.setErr(originalErr)
+
+        Assertions.assertThat(errContent.toString()).contains("Invalid input file for CSVExporter, stopping execution...")
+    }
+
+    @Test
+    fun `should fail to create output for invalid and valid file as input source`() {
+        val validInputFilePath = "src/test/resources/input_valid_1.cc.json"
+        val invalidInputFilePath = "filePathDoesNotExist.cc.json"
+
+        mockkObject(InputHelper)
+        every {
+            InputHelper.isInputValid(any(), any())
+        } returns false
+
+        System.setErr(PrintStream(errContent))
+        CommandLine(CSVExporter()).execute(validInputFilePath, invalidInputFilePath)
+        System.setErr(originalErr)
+
+        Assertions.assertThat(errContent.toString()).contains("Invalid input file for CSVExporter, stopping execution...")
+    }
+
+    @Test
+    fun `should fail to create output for folder as input source`() {
```

Comment:
By reading the test name, I am not sure what you want to test.
Maybe like this?

```suggestion
    fun `should fail with invalid input file message when path to a folder is given`() {
```